django/core/paginator.py

```diff
@@ -72,6 +75,8 @@ def page(self, number):
         top = bottom + self.per_page
         if top + self.orphans >= self.count:
             top = self.count
+        if self._slice_by:
+            return self._get_page(self.object_list.filter(pk__in=self.object_list_ids[bottom:top]), number, self)
```

Comment:
This should be a filter by `_slice_by__in` and not `pk__in`.

Reply:
No that is completely wrong, the idea is to check if primary key exists in a list of primary keys `object_list_ids` and it is not for any other type of columns. It is just for primary key.
Since different people use different naming for primary keys in their model so we need to specify that by  `_slice_by`.

Reply:
> Since different people use different naming for primary keys in their model so we need to specify that by `_slice_by`.

I don't understand your reasoning, why not use `'pk'` in your `.values` call then? The way you defined your API any field can be specified for `slice_by` which I thought has merit because if your queryset is ordered by anything but `pk`, e.g. another unique field, then it might be more efficient to retrieve rows by this field.

Reply:
I have renamed these variable, function (constructor) argument.